iron-list.html

```diff
@@ -817,6 +817,10 @@
           parseInt(styles['padding-top'], 10);
       this._isRTL = Boolean(styles.direction === 'rtl');
       this._viewportWidth = this.$.items.offsetWidth;
+      var viewportHeight = this._scrollTargetHeight;
+      if (viewportHeight !== this._viewportHeight) {
+        this._resetScrollPosition(this.scrollTarget.scrollTop);
+      }
```

Comment:
These lines could be on the chopping block for this PR, they are a little tricky. With the additional `requestAnimationFrame`s in the unit tests then this code isn't needed. However, I'm not sure if that approach more accurately recreates real life usage in the unit test context, or if acts as a work around to this code being absent.

Originally, I meant to submit this PR without this code, but with the phantom test failures I put it back in to see if I could get them to pass. With the addition of `list.style.width = '320px';` below, removing this could go either way. Thoughts?